### DISCUSSION TITLE
Does Uno Platform support native mobile widgets?

### DISCUSSION THREAD

**Discussion by cloud-aspect** [unoplatform/uno#8693] [Q&A]
I'm looking to start a new cross-platform project and currently researching options for language/frameworks/tooling, does Uno Platform Support both Android and iOS home screen widgets? Is it also possible to create such analogous widgets for native desktop platforms, e.g. [Microsoft Sticky Notes](https://apps.microsoft.com/store/detail/microsoft-sticky-notes/9NBLGGH4QGHW?hl=en-us&gl=US) and applications in this style for macOS and Linux?

I am continuing my investigation in the background, but wanted to drop this question here since I could not find answers to this quickly

Thank you.

**Comment by cloud-aspect** [comment]
The correct assumption seems to be yes for widgets because Uno is built on Xamarin which supports, for the most part, all the Android and iOS APIs, right?

In terms of doing something like Microsoft Sticky Notes, I would assume also yes, since we have complete support for all .NET libraries for Windows GUI applications. How does Linux and macOS support work? Still researching.

**Reply by jeromelaban** [reply]
As Uno apps are Xamarin.iOS apps, you can use all the APIs provided by iOS, including this: https://docs.microsoft.com/en-us/xamarin/ios/platform/search/widgets.

**Reply by cloud-aspect** [reply]
Fantastic, I would assume same for Android

**Comment by cloud-aspect** [comment]
BTW @jeromelaban I found Uno via Software Engineer Radio Episode 508, and was very impressed! I don't know C# or .NET (I have a background in Java/Spring backends and JS/HTML/CSS/React frontends), but I was moved by your discussion, and am happy to learn to be able to use Uno.

A lot of people have said that Uno is more geared to experienced .NET developers who wish to use their already established skill set to build cross-platform, and otherwise people tout competing frameworks like React Native, Flutter, etc. However, one big advantage Uno seems to have that people overlook is the maturity of the core interop layer, Xamarin, which supports virtually all native APIs, AND you can target web assembly, which I am convinced is the direction the web is going long term.
